tests/yam/agama/agama_headless.pm

```diff
@@ -0,0 +1,56 @@
+## Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Run Agama headless installation
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base Yam::Agama::agama_base;
+use strict;
+use warnings;
+use Carp qw(croak);
+use testapi qw(
+  get_var
+  script_run
+  script_output
+  assert_script_run
+  upload_logs
+  select_console
+  console
+);
+use Utils::Architectures qw(is_s390x is_ppc64le);
+use Utils::Backends qw(is_svirt);
+use utils qw(systemctl);
+use power_action_utils 'power_action';
+
+sub run {
+    my $self = shift;
+
+    # verify we are in muti-user target.
+    assert_script_run('systemctl list-units --type=target | grep multi-user.target');
+    systemctl('get-default', ignore_failure => 1);
+    systemctl('status graphical.target', ignore_failure => 1);
+    systemctl('status x11-autologin.service', ignore_failure => 1);
+    systemctl('status multi-user.target', ignore_failure => 1);
+
+    # check there is not instance of the browser open
+    assert_script_run('! pgrep -x \'firefox|chrome\' > /dev/null');
```

Comment:
could you please show here output of this to compare headless with normal one, headful? how does it look the failure, is clear, I don't see any message indicating the tester/user what happened. In general, please check that each command has a reasonable error.